pkg/controller/jobframework/interface.go

```diff
@@ -36,11 +36,21 @@ import (
 	"sigs.k8s.io/kueue/pkg/util/maps"
 )
 
-// GenericJob if the interface which needs to be implemented by all jobs
+// BaseJob is the interface which needs to be implemented by all jobs
 // managed by the kueue's jobframework.
-type GenericJob interface {
+type BaseJob interface {
```

Comment:
I do not want to relax the interface. Why do we not implement WorkloadPriority mechanism in LWS reconciler?
This interface is exposed to external. So, just relaxing is a bit dangerous.

Reply:
However, LeaderWorkerSet is not a GenericJob.

Reply:
Yes. So we can expose only utilfunctions for integration Jobs.

Reply:
@tenzen-y I know were you are coming from - it was also my question on slack, but I think without this interface we need more refactoring. Looking into options.

Reply:
Actually, let's take a backup of this work, and let's try the other approach without the new interface

https://github.com/kubernetes-sigs/kueue/pull/4711/files#r2005994624